### DISCUSSION TITLE
Question about the MaskMap

### DISCUSSION THREAD

**Discussion by aiinibxl** [iiasa/CWatM#58] [Q&A] [answered]
Hello everyone~

I have a problem here. If I use gauge coordinates as MaskMap like this:
<img width="222" alt="1" src="https://user-images.githubusercontent.com/96671336/209764164-8ad11596-25b7-472a-8651-78f5ad32732f.png">

the discharge output is correct, like this:
<img width="559" alt="2" src="https://user-images.githubusercontent.com/96671336/209765171-2b65ec59-6f4e-478a-8282-c58ceeccad39.png">

However, if I use a tif file as MaskMap like this:
<img width="413" alt="3" src="https://user-images.githubusercontent.com/96671336/209765469-df40e726-64db-4c0a-8867-7bf1a9f77d2e.png">

the discharge output at the same station become quite small, like this:
<img width="232" alt="4" src="https://user-images.githubusercontent.com/96671336/209765564-9f4501fe-c953-4091-96b2-d361e0c59ef9.png">

I don't know where the problem is, the tif file I use can cover the drainage area above the  gauge coordinates. Hope that someone can give me some advice. Thank you very much!

Lili

**Comment by dof1985** [comment]
Hi Lili

I am not sure about the mask map you are using, but I can provide some hints for you to explore this issue. 

First, just typing these coordinates in GoogleMaps put me at the Yangtze outlet. If you've aimed to model the Yangtze - so the discharge from CWatM generated mask map (top one) makes sense. 

Second, it would be easier if you provide an image of your mask map including some metadata, e.g. resolution, extent, and geographic projection. 

Good luck

**Reply by aiinibxl** [reply]
Hi Fridman

Thank you for your kind reply.

Actually when I use the coordinates as MaskMap, CWatM can generate maskmap correctly and the simulated discharge is reasonable.
The problem comes out when I use the image as MaskMap. I want to model the whole Chinese mainland, so I use 'china2.tif' as maskmap. And I also want to get the discharge at some specific points, for example, the outlet of Yangtze, so I type the point coordinates in 'Gauges = ', then the simulated discharge become quite small.
I think 'china2.tif' includes the metadata you've mentioned, like resolution, extent, and geographic projection. And I also compare it with the image provided in the tutorial 'rhine5min.tif', I think there is no big difference between them. So I don't know where the problem is.

Looking forward to your reply if you have some more advice~

Lili

**Reply by dof1985** [reply]
'I think 'china2.tif' includes the metadata you've mentioned, like resolution, extent, and geographic projection. '

**I mean that you can share here the metadata of the mask map.**

**Reply by aiinibxl** [reply]
This is the metadata, and I also upload 'china2.tif' in the zip file.

<img width="289" alt="11" src="https://user-images.githubusercontent.com/96671336/210074728-1559dc15-4fe2-4084-a774-3c141e6f9c23.png">
<img width="242" alt="22" src="https://user-images.githubusercontent.com/96671336/210074744-b13984c7-c929-40f4-b072-45637baca8b0.png">
[china2.zip](https://github.com/iiasa/CWatM/files/10324881/china2.zip)

**Comment by dof1985** [comment, marked as answer]
Looks weird. So usually, these types of discrepancies are related to choosing the wrong outlet, e.g., off the main river. 
Maybe the mask map is a bit shifted relative to other input data (e.g., ldd). Since the low discharge values show up when the catchment is pre-given - this just might be that you need to modify your guage location so that it coincides with the river. 

I guess it would be best to just resample your mask map to the ldd with the nearest neighbor method.

**Reply by aiinibxl** [reply]
Thank you very much for your advice.  I will try to resample the mask map.

**Reply by aiinibxl** [reply]
Thank you for your valuable advice, and I've finally solved the problem! It was precised because my mask map was not consistent with other input data (e.g., ldd) , and when I extract my maskmap from ldd, the model can function well.

**Reply by Abicula** [reply]
> Thank you for your valuable advice, and I've finally solved the problem! It was precised because my mask map was not consistent with other input data (e.g., ldd) , and when I extract my maskmap from ldd, the model can function well.


How did you extract the maskmap from ldd?

**Reply by aiinibxl** [reply]
> > Thank you for your valuable advice, and I've finally solved the problem! It was precised because my mask map was not consistent with other input data (e.g., ldd) , and when I extract my maskmap from ldd, the model can function well.
> 
> How did you extract the maskmap from ldd?

I use the multidimension tools in ArcGIS to import ldd as a raster file, then I use a shapefile covering my study area to extract this raster file.
